packages/ozone/src/mod-service/index.ts

```diff
@@ -791,6 +791,38 @@ export class ModerationService {
     return result
   }
 
+  applyTagFilter = (
+    builder: SelectQueryBuilder<any, any, any>,
+    tags: string[],
+  ) => {
+    const { ref } = this.db.db.dynamic
+    // Build an array of conditions
+    const conditions = tags.map((tag) => {
+      if (tag.includes('&&')) {
+        // Split by '&&' for AND logic
+        const subTags = tag
+          .split('&&')
+          // Make sure spaces on either sides of '&&' are trimmed
+          .map((subTag) => subTag.trim())
+          // Remove empty strings after trimming is applied
+          .filter(Boolean)
+        return sql`(${sql.join(
+          subTags.map(
+            (subTag) =>
+              sql`${ref('moderation_subject_status.tags')} ? ${subTag}`,
+          ),
+          sql` AND `,
+        )})`
```

Comment:
Do we ignore empty queries ? Or maybe even throw an error ?

```suggestion
        // False is neutral in a union ("OR")
        if (!subTags.length) return sql`FALSE`

        return sql`(${sql.join(
          subTags.map(
            (subTag) =>
              sql`${ref('moderation_subject_status.tags')} ? ${subTag}`,
          ),
          sql` AND `,
        )})`
```

Reply:
good catch! handled these edge cases through boolean filters.